JAGPropertyConverter/JAGPropertyConverter.m

```diff
@@ -424,20 +435,39 @@ - (void) setPropertiesOf: (id) object fromDictionary: (NSDictionary*) dictionary
     for (NSString *dictKey in dictionary) {
         
         NSString *key = dictKey;
-        
-        property = [JAGPropertyFinder propertyForName: key inClass:[object class] ];
+        BOOL isKeyPath;
+        NSString *remainingKeyPath = @"";
+
+        property = [JAGPropertyFinder propertyForName: key inClass:[object class]];
         
         if (!property) {
             // when enabled, convert to camelcase and try again fetching property
             if (self.enableSnakeCaseSupport) {
                 key = [key asCamelCaseFromUnderscore];
-                property = [JAGPropertyFinder propertyForName: key inClass:[object class] ];
+                property = [JAGPropertyFinder propertyForName: key inClass:[object class]];
             }
             // try custom mapping after snake case converting
-            if (!property && customMapping[key]) {
-                key = customMapping[key];
-                property = [JAGPropertyFinder propertyForName: key inClass:[object class] ];
+            if (!property) {
+                if (customMapping[key]) {
+                    key = customMapping[key];
+
+                    property = [JAGPropertyFinder propertyForName: key inClass:[object class] ];
+                }
+
+                // Check if it is a keypath and get the first segment
+                if (!property) {
+                    NSRange rangeOfFirstDot = [key rangeOfString:@"."];
+                    isKeyPath = rangeOfFirstDot.location != NSNotFound;
+
+                    if (isKeyPath) {
+                        remainingKeyPath = [key substringFromIndex:rangeOfFirstDot.location + 1];
+                        key = [key substringToIndex:rangeOfFirstDot.location];
+
+                        property = [JAGPropertyFinder propertyForName: key inClass:[object class]];
```

Comment:
i think this won't work for this use case: mapping = { modelProperty.modelProperty.stringProperty : "null" }

this would result in testModel.modelProperty == nil, which we don't want.